server/src/main/java/org/apache/iotdb/db/engine/merge/manage/MergeResource.java

```diff
@@ -68,9 +68,21 @@
   private boolean cacheDeviceMeta = false;
 
   public MergeResource(List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles) {
-    this.seqFiles = seqFiles.stream().filter(TsFileResource::isClosed).collect(Collectors.toList());
+    this.seqFiles = seqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted())
+        .collect(Collectors.toList());
     this.unseqFiles =
-        unseqFiles.stream().filter(TsFileResource::isClosed).collect(Collectors.toList());
+        unseqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted())
+            .collect(Collectors.toList());
+  }
+
+  public MergeResource(List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles,
+      long timeBound) {
+    this.seqFiles =
+        seqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted()
+            && res.stillLives(timeBound)).collect(Collectors.toList());
+    this.unseqFiles =
+        unseqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted()
+            && res.stillLives(timeBound)).collect(Collectors.toList());
```

Comment:
Every time the function `MergeResource` is invoked, two new class will be created because the `timeBound` in lambda express may be different among different invocations.
It's better to write a class to implement the interface Predicate<T> and maintain a field named `timeBound`. And set the field to different value in the function `MergeResource`.

Reply:
Yes, but considering the frequency of merge (at least every tens of minutes), such improvement is negligible.
Still, I replace the lambda with a method reference, please have a look.